lib/dialects/abstract/query.js

```diff
@@ -514,7 +514,19 @@ AbstractQuery.prototype.isUpdateQuery = function() {
 
 AbstractQuery.prototype.handleSelectQuery = function(results) {
   var result = null;
-
+  // Map raw field names using a map of aliased fields
+  if (this.options.aliasMap) {
+    var aliasMap = this.options.aliasMap;
+    results = Utils._.map(results, function(result) {
+      return Utils._.reduce(aliasMap, function(result, alias, field) {
+        if (field in result) {
+          result[alias] = result[field];
+          delete result[field];
```

Comment:
`result[field] = undefined` is a bit quicker, can you check if it produces the same result?